client/notifications/notification-reducer.ts

```diff
@@ -1,25 +1,23 @@
-import { List, Record, Set } from 'immutable'
+import { List, Map, Record } from 'immutable'
 import { assertUnreachable } from '../../common/assert-unreachable'
 import { EMAIL_VERIFICATION_ID, Notification, NotificationType } from '../../common/notifications'
 import keyedReducer from '../reducers/keyed-reducer'
 
 export interface NotificationRecordBase {
-  type: NotificationType
-  /**
-   * A string that is unique if the notification is unique. For locally-generated notifications
-   * that should never show duplicates, this might be a constant string. For server-generated ones
-   * that have multiple of the same type (e.g. chat notifications, party invites, etc.) these are
-   * generated by the server.
-   */
   id: string
-  unread: boolean
+  type: NotificationType
+  read: boolean
+  createdAt: number
+  local?: boolean
 }
 
 export class EmailVerificationNotificationRecord
   extends Record({
-    type: NotificationType.EmailVerification as typeof NotificationType.EmailVerification,
     id: EMAIL_VERIFICATION_ID,
-    unread: true,
+    type: NotificationType.EmailVerification as typeof NotificationType.EmailVerification,
+    read: false,
+    createdAt: 0,
+    local: true,
```

Comment:
I'm not sure what I'm doing wrong, but I get this error when I try to do that:

![image](https://user-images.githubusercontent.com/4757944/118343273-3a844b80-b528-11eb-96a2-a1ebf2baab69.png)

Reply:
Yeah, that's weird. Testing it out in TS Playground it seems it works as `true as true`. Odd that it is different from string literals but 🤷

Reply:
Using `true as true` gives me an eslint error though:

![image](https://user-images.githubusercontent.com/4757944/118356550-1d766980-b576-11eb-845e-12a8d8707e49.png)

And if I change that to `true as const`, then I get another error I don't know how to fixx:

![image](https://user-images.githubusercontent.com/4757944/118356579-38e17480-b576-11eb-8761-f5ba69b1496f.png)

Reply:
huh, was not aware that was a thing, guess it got added after I last used TS significantly :) ( https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions )

I think it may need a change to `NotificationRecordBase`, let me play around with it

Reply:
Ah okay, actually straightforward:

In `common/notifications.ts`, you need to update `EmailVerificationNotification` to specify that `local` will always be `true` (instead of allowing it to be any `boolean` value). e.g.

```ts
export interface EmailVerificationNotification extends BaseNotification {
  type: typeof NotificationType.EmailVerification
  local: true
}
```

The error you were seeing above is because when TS has narrowed the type to just an `EmailVerificationNotification`, it still allowed for `local` to be `false` or `undefined` because of the base interface.